frontend/packages/dev-console/src/components/topology/__tests__/TopologyPage.spec.tsx

```diff
@@ -0,0 +1,95 @@
+
+let topologyProps: TopologyPageProps;
+let renderTopologyProps: RenderTopologyProps;
+
+describe('Topology page tests', () => {
+  beforeEach(() => {
+    topologyProps = {
+      activeApplication: 'topology-test',
+      knative: false,
+      cheURL: 'che.openshift.io',
+      serviceBinding: false,
+      match: {
+        params: {
+          name: 'topology-test',
+        },
+        isExact: true,
+        path: '',
+        url: '',
+      },
+    };
+
+    renderTopologyProps = {
+      data: topologyData,
+      loaded: true,
+      loadError: '',
+      serviceBinding: false,
+    };
+  });
+
+  it('should render topology list page', () => {
+    topologyProps.match.path = '/list';
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    expect(wrapper.find(AsyncComponent).exists()).toBe(true);
+  });
+
+  it('should render topology graph page', () => {
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    expect(wrapper.find(ConnectedTopologyDataController).exists()).toBe(true);
+  });
+
+  it('should render projects list page', () => {
+    topologyProps.match.params.name = '';
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    expect(wrapper.find(ProjectListPage).exists()).toBe(true);
+  });
+
+  it('should render view shortcuts button on topology page toolbar', () => {
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    const namespacesPageWrapper = wrapper.find(NamespacedPage).shallow();
+    expect(namespacesPageWrapper.find(Button).props().children).toEqual('View shortcuts');
+  });
+
+  it('should not render view shortcuts button on topology list page toolbar', () => {
+    topologyProps.match.path = '/list';
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    const namespacesPageWrapper = wrapper.find(NamespacedPage).shallow();
+    expect(namespacesPageWrapper.find(Button).exists()).toBe(false);
+  });
+
+  it('should show the topology icon when on topology list page', () => {
+    topologyProps.match.path = '/list';
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    const namespacesPageWrapper = wrapper.find(NamespacedPage).shallow();
+    expect(namespacesPageWrapper.find(Tooltip).props().content).toBe('Topology View');
+    expect(namespacesPageWrapper.find(Link).props().to).toBe('/topology/ns/topology-test');
+  });
+
+  it('should show the topology list icon when on topology page', () => {
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    const namespacesPageWrapper = wrapper.find(NamespacedPage).shallow();
+    expect(namespacesPageWrapper.find(Tooltip).props().content).toBe('List View');
+    expect(namespacesPageWrapper.find(Link).props().to).toBe('/topology/ns/topology-test/list');
+  });
+
+  it('should render topology when workload is loaded', () => {
+    const RenderTopology = React.createElement(renderTopology, renderTopologyProps);
```

Comment:
You're still treating this as if it was a component you could render. It's a function, not a functional component. I think this action is a misnomer. Just invoke the method and use a fake component to wrap the output for testing:

```
  it('should render topology when workload is loaded', () => {
    const Component = () => renderTopology(renderTopologyProps);
    const wrapper = shallow(<Component />);
    expect(wrapper.find(StatusBox).exists()).toBe(true);
    expect(wrapper.find(Topology).exists()).toBe(true);
  });
```

Reply:
Fix this.